[android] Fix powl test in Android ARMv7/AArch64.

drodriguez opened the pull request:
In Android ARMv7 there was no alias defined for CLongDouble, so
importing anything with long double was failing. Defining the alias
makes the test pass.

AAPCS64 defines long double as 128 bits, which is still unsupported by
Swift (SR-9072). Disable this particular test for Android and Linux in
AArch64 to not fail (but still test all the other tests).

drodriguez committed:
[android] Fix powl test in Android ARMv7/AArch64.

In Android ARMv7 there was no alias defined for CLongDouble, so
importing anything with long double was failing. Defining the alias
makes the test pass.

AAPCS64 defines long double as 128 bits, which is still unsupported by
Swift (SR-9072). Disable this particular test for Android and Linux in
AArch64 to not fail (but still test all the other tests).

compnerd commented:
I think that we should add the 128-bit FP support.  It is part of the AAPCS64 EABI, so this impacts more than just android.

drodriguez commented:
Agree, but that will imply implementing first and going through the Swift Evolution process second (since it changes the public API). From my understanding, it might apply to Linux AArch64, since x86/_64 `long double` are 80 bits (which are already supported). I can open a JIRA to track, but I really think that this is a good solution to get this test passing at the moment.

drodriguez commented:
/cc @stephentyrone We think we should ask for your opinion. Implementing `Float128` is possible. I have to implement `(U)Int128` first for that. It will take a little bit more time. Since we cannot xfail the test because the compilation is what’s failing, I think disabling the test for the time being is a good measure.

stephentyrone commented:
Adding fp128 support is definitely the right long-term thing to do, but it's fairly low-priority, and a decent amount of work. If someone wants to hack on it, I'm happy to give pointers.

Disabling the test for now is fine. Is this really the only one that needs to be switched off?

drodriguez commented:
@stephentyrone: I started the support for `U`/`Int128` in https://github.com/drodriguez/swift/commit/dd20b663bcc091119cf317c67013dae3c135b963, since it is needed for representing the significand of a fp128. I did some exploration on fp128, but I will appreciate those pointers, in case I get some time to hack on it (or any feedback on the 128 bits integers changes, if you have them. I can open a WIP PR, if you prefer).

compnerd commented:
Can you please create a SR for that and reference that?

drodriguez commented:
👍 https://bugs.swift.org/browse/SR-9072

drodriguez commented:
This is probably not going to be the final PR. I haven’t have time to implement it, but support for fp128 will be the right way of fixing this. I only have this open as a reminder that I need to work on that.

drodriguez commented:
@compnerd: reworked this old PR for the `cfuncs_parse` test failure. For Android ARMv7 the test is fixed (it was missing a typealias for `CLongDouble`). For Android AArch64 the test is disabled, since the fp128 is not still there. I will try to work on it, but it might take a while.

@futurejones: I added Linux there to avoid the same problem, as you asked.

futurejones commented:
@drodriguez thanks!

stephentyrone commented:
@swift-ci please test
